modules/core/output_modules.php

```diff
@@ -1645,7 +1645,99 @@ class Hm_Output_content_section_end extends Hm_Output_Module {
      * Closes the main tag opened in Hm_Output_content_section_start
      */
     protected function output() {
-        return '</div></main>';
+        $share_folder_modal = '<div class="modal fade" id="shareFolderModal" tabindex="-1" aria-labelledby="shareFolderModalLabel" aria-hidden="true">';
```

Comment:
This class is a section_end - please put all that modal code in one or more separate classes and include those in the setup.php file where appropriate.

Reply:
new class added. @kroky i hope this PR is ready.